### PR TITLE
create, sync & share n files in the root directory

### PR DESCRIPTION
**Pull request by mrow4a**
This test will create, sync & share n files in the root directory. It will print out the following details for each files:

````
DATE                                                        FILENAME                           N  SIZE SYNC-T   SHARE-T

2016-12-04 23:49:25,959 - INFO - sharer - SMASHED-SYNC-SHARE-214021772441fbe455b29ee89baa1764-100-1000=0.430163-0.422634
````

It means, it will show how long it took to sync a specific file, and also how long it took to share this file. 

It will repeat the operation for each of N files.

### CODE REVIEW

**Comment by felixboehm** [comment]
That looks good. Can you used "named steps" in smashbox tests, so that steps would be easy to understand in the top of the file.

```
STEP_PREPARATION = 1
STEP_CREATE_FILE = 2
STEP_SHARE_FILE = 3
STEP_SYNC_FILE = 4 
```
...
What do you think?

**Comment by mrow4a** [comment]
@felixboehm Done

**Comment by mrow4a** [comment]
@jvillafanez Can I merge it?

**Comment by felixboehm** [comment]
👍  merge it.

improvements as second step:
- Allow larger n (fast) to study the limits
- run test until request time escalates - say 10s request time.

**Comment by mrow4a** [comment]
@felixboehm Merge with changes?

File: python/smashbox/multiprocessing_engine.py [hunk 1]
```diff
@@ -113,16 +113,22 @@ def supervisor(steps):
         if _smash_.DEBUG:
             log('start',_smash_.supervisor_step.value,_smash_.steps)
 
+        _smash_.N_STEPS = max(
+            int(config.get('smashbox_steps_limit', -1)), #this can be configured using [-o smashbox_steps_limit=N]
+            _smash_.N_STEPS)
+
```
Comment (mrow4a) [inline python/smashbox/multiprocessing_engine.py:119]: @moscicki What do you think ? Could we define number of steps in config file? Use case is that you want to sync many many files, and have a synchronisation between them for each step.

File: lib/performance/test_nSharedFiles.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+    for i in range(nfiles):
+        #STEP 1: sync down the shared file (PROPFIND and GET) and measure the time
+        time0 = time.time()
+        run_ocsync(d,user_num=shareReceiverNum)
+        time1 = time.time()
+        logger.info("SMASHED-RECEIVER-%i-%i=%f" % (nfiles, filesize, time1 - time0))
+        next_step = next_step + 2
+        step(next_step, 'Synced down %s' % i)
+
```
Comment (mrow4a) [inline lib/performance/test_nSharedFiles.py:152]: @moscicki Here is the use case, I am doing step() function in the for loop

**Comment by PVince81** [comment]
@mrow4a are the additional points of enough value to hold this PR ? Else merge this and address separately so we can at least have more coverage

**Comment by mrow4a** [comment]
If the topic will reapear, will reopen this

**Pull request closed without merging**